modules/virtual_network_gateway/main.tf

```diff
@@ -2,78 +2,78 @@
+      aad_issuer    = vpn_client_configuration.value.aad_issuer
       dynamic "root_certificate" {
-        for_each = try(vpn_client_configuration.value.root_certificate, null) != null ? { for t in vpn_client_configuration.value.root_certificate : t.name => t } : {}
+        for_each = vpn_client_configuration.value.root_certificate != null ? { for t in vpn_client_configuration.value.root_certificate : t.name => t } : {}
         content {
           name             = root_certificate.value.name
           public_cert_data = root_certificate.value.public_cert_data
         }
       }
       dynamic "revoked_certificate" {
-        for_each = try(vpn_client_configuration.value.revoked_certificate, null) != null ? { for t in vpn_client_configuration.value.revoked_certificate : t.name => t } : {}
+        for_each = vpn_client_configuration.value.revoked_certificate != null ? { for t in vpn_client_configuration.value.revoked_certificate : t.name => t } : {}
         content {
           name       = revoked_certificate.value.name
           thumbprint = revoked_certificate.value.thumbprint
         }
       }
-      radius_server_address = try(vpn_client_configuration.value.radius_server_address, null)
-      radius_server_secret  = try(vpn_client_configuration.value.radius_server_secret, null)
-      vpn_client_protocols  = try(vpn_client_configuration.value.vpn_client_protocols, null)
-      vpn_auth_types        = try(vpn_client_configuration.value.vpn_auth_types, null)
+      radius_server_address = vpn_client_configuration.value.radius_server_address
+      radius_server_secret  = vpn_client_configuration.value.radius_server_secret
+      vpn_client_protocols  = vpn_client_configuration.value.vpn_client_protocols
+      vpn_auth_types        = vpn_client_configuration.value.vpn_auth_types
     }
   }
 
   dynamic "bgp_settings" {
     for_each = [var.local_bgp_settings]
     content {
-      asn = try(bgp_settings.value.asn, null)
+      asn = bgp_settings.value.asn
       dynamic "peering_addresses" {
-        for_each = try(bgp_settings.value.peering_addresses, {})
+        for_each = bgp_settings.value.peering_addresses
         content {
-          ip_configuration_name = try(peering_addresses.key, null)
-          apipa_addresses       = [for i in try(peering_addresses.value.apipa_addresses, []) : var.azure_bgp_peers_addresses[i]]
-          default_addresses     = try(peering_addresses.value.default_addresses, null)
+          ip_configuration_name = peering_addresses.key
+          apipa_addresses       = [for i in peering_addresses.value.apipa_addresses : var.azure_bgp_peers_addresses[i]]
+          default_addresses     = peering_addresses.value.default_addresses
         }
       }
-      peer_weight = try(bgp_settings.value.peer_weight, null)
+      peer_weight = bgp_settings.value.peer_weight
     }
   }
 
   dynamic "custom_route" {
     for_each = var.custom_route
```

Comment:
this will brake when you assign null to `var.custom_route`, see comments in `variables.tf` about `nullable`

Reply:
fixed, thanks